### PR TITLE
Refactor methods to run backend test tasks

### PR DESCRIPTION
**Pull request by eduardoj**
Reduce duplication offenses reported by codeclimate.

Also, remove a method which was never used.

### CODE REVIEW

File: src/api/app/lib/backend/test/tasks.rb [hunk 1]
```diff
@@ -3,39 +3,15 @@ class Test
...
       def run_scheduler(arch)
-        Rails.logger.debug { "RUN_SCHEDULER #{arch}" }
-        perlopts = "-I#{Rails.root}/../backend -I#{Rails.root}/../backend/build"
-        IO.popen("cd #{backend_config}; exec perl #{perlopts} ./bs_sched --testmode #{arch}") do |io|
-          # just for waiting until scheduler finishes
-          io.each { |line| Rails.logger.debug { "scheduler(#{arch}): #{line.strip.chomp}" } if line.present? }
-        end
-      end
-
-      def run_dispatcher
```
Comment (dmarcoux) [inline src/api/app/lib/backend/test/tasks.rb:14]: This method is now gone, is this what we want? I don't think so, could you confirm?

File: src/api/app/lib/backend/test/tasks.rb [hunk 2]
```diff
@@ -49,6 +25,16 @@ def run_admin(args)
         end
         ret
       end
+
+      private
+
+      def run_backend_server(name:, command:)
```
Comment (adrianschroeter) [inline src/api/app/lib/backend/test/tasks.rb:31]: this name is highly confusing, could you use something like "single_scheduler_evaluation_run" or something like that?

Comment (adrianschroeter) [reply src/api/app/lib/backend/test/tasks.rb:31]: hm, or something like "execute_backend_daemon".

but "backend_server" are many processes for me, not a single temporary execution.

Comment (eduardoj) [reply src/api/app/lib/backend/test/tasks.rb:31]: "single_scheduler_evaluation_run" could be more confusing. This method not only runs a scheduler. I'm open to more ideas.

Comment (eduardoj) [reply src/api/app/lib/backend/test/tasks.rb]: I renamed the method to `execute_backend_daemon`.

**Comment by adrianschroeter** [comment]
On Dienstag, 28. November 2023, 11:31:35 CET Eduardo J. wrote:
> @eduardoj commented on this pull request.
> 
> 
> 
> > @@ -49,6 +25,16 @@ def run_admin(args)
>          end
>          ret
>        end
> +
> +      private
> +
> +      def run_backend_server(name:, command:)
> 
> I renamed the method to `execute_backend_daemon`.

sounds good

-- 

Adrian Schroeter ***@***.***>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH, Frankenstraße 146, 90461 Nürnberg, Germany
(HRB 36809, AG Nürnberg) Geschäftsführer: Ivo Totev

**Pull request merged**